### PR TITLE
add hide java wizard toggle

### LINKED ISSUE #506: [Suggestion] Add "Don't show again" option on java startup setup
### Role

I play modded minecraft

### Suggestion

Add "Don't show again" option on java startup setup

### Benefit

As all my java binaries written to settings of minecraft versions, I don't  java installed globally. So every time I start polyMC - this window show up.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

image
image

### PR DESCRIPTION
**Pull request by d-513**
Closes #506 

Done a bit differently though for consistency

### CODE REVIEW

**Comment by d-513** [comment]
@Scrumplex I used qt designer as you suggested, but it did a werid diff. Should I do it by hand or is it fine?

**Comment by Scrumplex** [comment]
> @Scrumplex I used qt designer as you suggested, but it did a werid diff. Should I do it by hand or is it fine?

Nope looks fine. Personally, I am not sure why it *sometimes* re-orders stuff.

**Review by Scrumplex — APPROVED** [review]

**Comment by d-513** [comment]
This is a gamer moment. Scrumplex approved PR first time without change request

**Review by txtsd — APPROVED** [review]

File: launcher/ui/pages/global/JavaPage.ui [hunk 1]
```diff
@@ -196,45 +167,84 @@
...
+          </item>
+          <item row="5" column="1">
+           <widget class="QCheckBox" name="skipJavaWizardCheckbox">
             <property name="toolTip">
-             <string>If enabled, the launcher will not check if an instance is compatible with the selected Java version.</string>
+             <string>If enabled, the launcher will not prompt you to choose a Java version if one isn't found.</string>
             </property>
             <property name="text">
-             <string>&amp;Skip Java compatibility checks</string>
+             <string>Skip Java Wizard</string>
```
Comment (Scrumplex) [inline launcher/ui/pages/global/JavaPage.ui]: Needs mnemonic
maybe
```suggestion
             <string>Skip Java &Wizard</string>
```
if W isn't use somewhere else on that page

**Comment by Scrumplex** [comment]
> This is a gamer moment. Scrumplex approved PR first time without change request

Thanks for pointing that out, I just looked again and found something

**Comment by d-513** [comment]
what tf is mnemonic anyways

File: launcher/ui/pages/global/JavaPage.ui [hunk 2]
```diff
@@ -196,45 +167,84 @@
...
+          </item>
+          <item row="5" column="1">
+           <widget class="QCheckBox" name="skipJavaWizardCheckbox">
             <property name="toolTip">
-             <string>If enabled, the launcher will not check if an instance is compatible with the selected Java version.</string>
+             <string>If enabled, the launcher will not prompt you to choose a Java version if one isn't found.</string>
             </property>
             <property name="text">
-             <string>&amp;Skip Java compatibility checks</string>
+             <string>Skip Java &Wizard</string>
```
Comment (Scrumplex) [inline launcher/ui/pages/global/JavaPage.ui]: I did epic cringe

```suggestion
             <string>Skip Java &amp;Wizard</string>
```

**Comment by Scrumplex** [comment]
> what tf is mnemonic anyways

When you press Alt+<the letter followed by the mnemonic marker> it focuses that element.

So for example if we say `Skip Java &amp;Wizard` (`Skip Java &Wizard`) Alt+W will focus the widget

**Comment by d-513** [comment]
ok

**Review by Scrumplex — APPROVED** [review]

**Review by timoreo22 — APPROVED** [review]

**Pull request merged**